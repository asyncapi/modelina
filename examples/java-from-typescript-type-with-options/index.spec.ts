const spy = jest.spyOn(global.console, 'log').mockImplementation(() => { return; });
import {generate} from './index';

describe(`Should be able to generate Java models from a TypeScript type file 
along with specific options`, () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('and should log expected output to console', async () => {
    await generate();
    //GenerateWithOptions is called 4x, so even though we expect 1 model, we double it
    expect(spy.mock.calls.length).toEqual(4);
    expect(spy.mock.calls[1]).toMatchSnapshot();
  });
});

