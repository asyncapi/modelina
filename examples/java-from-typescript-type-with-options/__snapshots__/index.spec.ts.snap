// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to generate Java models from a TypeScript type file 
along with specific options and should log expected output to console 1`] = `
Array [
  "public class OuterData {
  private Object inner;
  private String name;
  private Map<String, Object> additionalProperties;

  public Object getInner() { return this.inner; }
  public void setInner(Object inner) { this.inner = inner; }

  public String getName() { return this.name; }
  public void setName(String name) { this.name = name; }

  public Map<String, Object> getAdditionalProperties() { return this.additionalProperties; }
  public void setAdditionalProperties(Map<String, Object> additionalProperties) { this.additionalProperties = additionalProperties; }
}",
]
`;
